app/controllers/announcements_controller.rb

```diff
@@ -1,8 +1,17 @@
 # frozen_string_literal: true
 class AnnouncementsController < ApplicationController
+  load_resource :announcement, class: GenericAnnouncement.name, only: :mark_as_read,
+                               id_param: :announcement_id
+
   add_breadcrumb :index, :announcements_path
 
-  def index
+  def index #:nodoc:
     @announcements = global_announcements.includes(:creator)
   end
+
+  # Marks the current GenericAnnouncement as read by the current user and responds without a body.
+  def mark_as_read
+    @announcement.mark_as_read! for: current_user
+    head :ok
```

Comment:
@allenwq @weiqingtoh I don't think authorization is necessary here. Also, I'm not sure if I need to handle the case where `mark_as_read!` fails. Any thoughts?